backend/infrahub/git/base.py

```diff
@@ -164,10 +161,10 @@ class InfrahubRepositoryBase(BaseModel, ABC):
 
     @property
     def sdk(self) -> InfrahubClient:
-        if self.client:
-            return self.client
+        if not self.client:
+            self.client = get_client()
 
-        return self.service.client
+        return self.client
```

Comment:
It could perhaps be a follow up PR but I think we should get rid of the `sdk` property.